substrate/frame/identity/src/lib.rs

```diff
@@ -1060,45 +1110,42 @@ pub mod pallet {
 		///   - When combined with the suffix of the issuing authority be _less than_ the
 		///     `MaxUsernameLength`.
 		#[pallet::call_index(17)]
-		#[pallet::weight(T::WeightInfo::set_username_for())]
+		#[pallet::weight(T::WeightInfo::set_username_for(if *use_allocation { 1 } else { 0 }))]
 		pub fn set_username_for(
 			origin: OriginFor<T>,
 			who: AccountIdLookupOf<T>,
 			username: Vec<u8>,
 			signature: Option<T::OffchainSignature>,
+			use_allocation: bool,
```

Comment:
Update fn rustdoc about `use_allocation` flag? Currently it only says authority must have an `allocation`.